### PR TITLE
[WIP] Example on using HTTP #839

### PR DESCRIPTION
**Pull request by JohnCashmore**
Example on using http to request data and how to test it.

Currently a WIP, will implement the tests at lunch, might be worth having a quick look over the current progress to see if you agree with how its been implemented.

#839

### CODE REVIEW

**Comment by JohnCashmore** [comment]
The test is now in place for the reworked name-list service, however there are some other issues with the other tests that need to now be resolved, I'll hold off doing this until @mgechev makes a decision on whether this is a good way to go, I am easy either way. Just trying to think of the stuff i've struggled to figure out whilst building the angular 2 app i am working on using the seed project.

**Comment by mgechev** [comment]
LGMT

**Comment by ludohenin** [comment]
for me too.

File: src/client/app/shared/services/name-list.service.ts [hunk 1]
```diff
@@ -1,12 +1,37 @@
...
-  ];
+  names:string[] = [];
+  private loading: Boolean = false;
+  constructor(private http: Http) {
+  }
+
+  requestNames() {
+      this.http.get('./assets/json/name-list.json')
+      .map(this.mapData)
+      .subscribe(this.updateNames, this.handleError);
```
Comment (ludohenin) [inline src/client/app/shared/services/name-list.service.ts]: Shouldn't `this.updateNames` be bound to `this` (this.updateNames.bind(this)) ?

Comment (JohnCashmore) [inline src/client/app/shared/services/name-list.service.ts]: To be honest i am not sure on that, i assumed scoping issues where handled by L51, but I am not certain on this.

**Comment by JohnCashmore** [comment]
I've updated the specs where required, i've had to reduce the homepage check slightly, anyone any ideas on how best to tackle this with the mocked namelist, or are you happy as it is?

**Comment by mgechev** [comment]
@JakePartusch mocking looks good to me.

**Comment by JohnCashmore** [comment]
I'll have to revisit this at some point but closing for now as I haven't got time at the moment to bring this up to date.

**Pull request closed without merging**